### PR TITLE
[shared-mempool] convert to using prost

### PR DESCRIPTION
**Pull request by bmwill**
Continue converting SharedMempool to use prost for protobuf

### CODE REVIEW

File: types/src/transaction.rs [hunk 1]
```diff
@@ -526,6 +526,24 @@ impl IntoProto for SignedTransaction {
     }
 }
 
+impl TryFrom<crate::proto::types::SignedTransaction> for SignedTransaction {
+    type Error = Error;
+
+    fn try_from(txn: crate::proto::types::SignedTransaction) -> Result<Self> {
+        let signed_txn = SimpleDeserializer::deserialize(&txn.signed_txn)?;
+        Ok(signed_txn)
```
Comment (metajack) [inline types/src/transaction.rs]: I think you can just return the `deserialize` result directly here.

File: network/src/proto/mempool.proto [hunk 2]
```diff
@@ -3,7 +3,7 @@
 
 syntax = "proto3";
 
-package network;
+package mempool;
```
Comment (metajack) [inline network/src/proto/mempool.proto]: Was this a bug you fixed? I don't quite understand why the package name would change, although it seems like it should be mempool not network.

File: network/src/validator_network/test.rs [hunk 3]
```diff
@@ -162,12 +166,14 @@ fn test_mempool_sync() {
...
-    let mut mempool_msg = MempoolSyncMsg::new();
-    mempool_msg.set_peer_id(dialer_peer_id.into());
+    let mut mempool_msg = MempoolSyncMsg::default();
+    mempool_msg.peer_id = dialer_peer_id.into();
     let sender = AccountAddress::new([0; ADDRESS_LENGTH]);
     let keypair = compat::generate_keypair(&mut rng);
-    let txn = get_test_signed_txn(sender, 0, keypair.0, keypair.1, None).into_proto();
-    mempool_msg.set_transactions(::protobuf::RepeatedField::from_vec(vec![txn.clone()]));
+    let txn: SignedTransaction = get_test_signed_txn(sender, 0, keypair.0, keypair.1, None)
+        .try_into()
```
Comment (metajack) [inline network/src/validator_network/test.rs]: Getting rid of `::protobuf::RepeatedField::from_vec` is so much better.

File: types/src/transaction.rs (hunk 1 above)
Comment (bmwill) [reply types/src/transaction.rs]: yep you're right

File: network/src/proto/mempool.proto (hunk 2 above)
Comment (bmwill) [reply network/src/proto/mempool.proto]: I honestly don't know what the package thing is for. It seems like rust-protobuf ignores it and prost expects the generated code to be in a module named after the package used in the .proto file

File: network/src/validator_network/test.rs (hunk 3 above)
Comment (bmwill) [reply network/src/validator_network/test.rs]: mhmm!

**Pull request closed without merging**